spec/system/new_topic_list_spec.rb

```diff
@@ -0,0 +1,444 @@
+# frozen_string_literal: true
+
+describe "New topic list", type: :system do
+  fab!(:user) { Fabricate(:user) }
+  fab!(:group) { Fabricate(:group, users: [user]) }
+  fab!(:category) { Fabricate(:category) }
+  fab!(:tag) { Fabricate(:tag) }
+
+  fab!(:new_reply) { Fabricate(:post).topic }
+  fab!(:new_topic) { Fabricate(:post).topic }
+  fab!(:old_topic) { Fabricate(:post).topic }
+
+  fab!(:new_reply_in_category) do
+    Fabricate(:post, topic: Fabricate(:topic, category: category)).topic
+  end
+  fab!(:new_topic_in_category) do
+    Fabricate(:post, topic: Fabricate(:topic, category: category)).topic
+  end
+  fab!(:old_topic_in_category) do
+    Fabricate(:post, topic: Fabricate(:topic, category: category)).topic
+  end
+
+  fab!(:new_reply_with_tag) { Fabricate(:post, topic: Fabricate(:topic, tags: [tag])).topic }
+  fab!(:new_topic_with_tag) { Fabricate(:post, topic: Fabricate(:topic, tags: [tag])).topic }
+  fab!(:old_topic_with_tag) { Fabricate(:post, topic: Fabricate(:topic, tags: [tag])).topic }
+
+  let(:topic_list) { PageObjects::Components::TopicList.new }
+  let(:tabs_toggle) { PageObjects::Components::NewTopicListToggle.new }
+
+  before do
+    sign_in(user)
+
+    [old_topic, old_topic_in_category, old_topic_with_tag].each do |topic|
+      TopicUser.update_last_read(user, topic.id, 1, 1, 1)
+    end
+
+    [new_reply, new_reply_in_category, new_reply_with_tag].each do |topic|
+      TopicUser.change(
+        user.id,
+        topic.id,
+        notification_level: TopicUser.notification_levels[:tracking],
+      )
+      TopicUser.update_last_read(user, topic.id, 1, 1, 1)
+      Fabricate(:post, topic: topic)
+    end
+  end
+
+  shared_examples "new list new topics and replies toggle" do
+    context "when the new new view is enabled" do
+      before { SiteSetting.experimental_new_new_view_groups = group.name }
+
+      it "shows all new topics and replies by default" do
+        visit("/new")
+
+        expect(topic_list).to have_topics(count: 6)
+        [
+          new_reply,
+          new_topic,
+          new_reply_in_category,
+          new_topic_in_category,
+          new_reply_with_tag,
+          new_topic_with_tag,
+        ].each { |topic| expect(topic_list).to have_topic(topic) }
+
+        expect(tabs_toggle.all_tab).to have_count(6)
+        expect(tabs_toggle.replies_tab).to have_count(3)
+        expect(tabs_toggle.topics_tab).to have_count(3)
+      end
+
+      it "the All tab is the default is the default tab" do
+        visit("/new")
+
+        expect(tabs_toggle.all_tab).to be_active
+        expect(tabs_toggle.replies_tab).to be_inactive
+        expect(tabs_toggle.topics_tab).to be_inactive
```

Comment:
I think these assertions can be combined with the test case above.

Reply:
done!